core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3089,13 +3089,14 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     submit(finalRdd, Array(0, 1), properties = new Properties())
 
     // Finish the first 2 shuffle map stages.
-    completeShuffleMapStageSuccessfully(0, 0, 2)
+    completeShuffleMapStageSuccessfully(0, 0, 2, Seq("hostA", "hostB"))
```

Comment:
This not only fixes the comment but the test codes. Mind elaborating this please?

Reply:
Yes. The background is that: Wenchen and I were reading the related code, but felt confused about why line 3098 says: "Executor lost on hostB, both of stage 0 and 1 should be reran.". 

After digging around, we found that stage0 completes in "hostA", "hostB" (the default hosts somewhere in the code). But it is not very obvious for reader to tell this.

So adding  Seq("hostA", "hostB") to reader to easily get this. I further updated the comments too.

Reply:
This change is not required.
Fetch failed is due to stage 1 partition on hostB going missing - it does not matter where stage 0 ran - it will be recomputed for indeterminate stages.

Reply:
Yes, this is not required. It is only added for better readability (as I mentioned, "hostA", "hostB" are the default hosts). In my opinion, having the value here, we don't need to go to read completeShuffleMapStageSuccessfully(), but only the code here, we can know what happen. Beside, it might be good to keep the two completeShuffleMapStageSuccessfully() consistent here (both having hostNames, or both not having hostNames)?  Let me know if you think I should delete Seq("hostA", "hostB") here. 

I added "In case no hostNames are provided, the tasks will progressively complete on hostA, hostB, etc." to completeShuffleMapStageSuccessfully().

Reply:
Enhancing documentation of `completeShuffleMapStageSuccessfully` is sufficient - we dont want to explicitly specify `hostNames` for all invocations of the method.

Reply:
Sounds good. Deleted the hostNames param.